pkg/kritis/crd/securitypolicy/securitypolicy_test.go

```diff
@@ -77,84 +79,68 @@ func Test_UnqualifiedImage(t *testing.T) {
-		Spec: v1beta1.ImageSecurityPolicySpec{
-			PackageVulnerabilityRequirements: v1beta1.PackageVulnerabilityRequirements{
-				MaximumSeverity: "BLOCKALL",
-				WhitelistCVEs:   []string{"l", "m", "c"},
-			},
-		},
-	}
+func Test_SeverityThresholds(t *testing.T) {
 	mc := &testutil.MockMetadataClient{
 		Vulnz: []metadata.Vulnerability{
-			{CVE: "l", Severity: "LOW"},
-			{CVE: "m", Severity: "MEDIUM"},
-			{CVE: "c", Severity: "CRITICAL"},
-		},
-	}
-	violations, err := ValidateImageSecurityPolicy(isp, testutil.QualifiedImage, mc)
-	if err != nil {
-		t.Errorf("error validating isp: %v", err)
-	}
-	if violations != nil {
-		t.Errorf("got unexpected violations: %v", violations)
-	}
-}
-
-func Test_BlockallFail(t *testing.T) {
-	isp := v1beta1.ImageSecurityPolicy{
-		Spec: v1beta1.ImageSecurityPolicySpec{
-			PackageVulnerabilityRequirements: v1beta1.PackageVulnerabilityRequirements{
-				MaximumSeverity: "BLOCKALL",
-				WhitelistCVEs:   []string{"m"},
-			},
-		},
-	}
-	mc := &testutil.MockMetadataClient{
-		Vulnz: []metadata.Vulnerability{{CVE: "l", Severity: "LOW"}},
-	}
-	violations, err := ValidateImageSecurityPolicy(isp, testutil.QualifiedImage, mc)
-	expected := []Violation{
-		{
-			Vulnerability: mc.Vulnz[0],
-			Violation:     ExceedsMaxSeverityViolation,
-			Reason:        ExceedsMaxSeverityViolationReason(testutil.QualifiedImage, mc.Vulnz[0], isp),
-		},
-	}
-	testutil.CheckErrorAndDeepEqual(t, false, err, violations, expected)
-}
-
-func Test_MaxSeverityFail(t *testing.T) {
-	isp := v1beta1.ImageSecurityPolicy{
-		Spec: v1beta1.ImageSecurityPolicySpec{
-			PackageVulnerabilityRequirements: v1beta1.PackageVulnerabilityRequirements{
-				MaximumSeverity: "MEDIUM",
-			},
+			{CVE: "l", Severity: "LOW", HasFixAvailable: true},
+			{CVE: "l_nofix", Severity: "LOW", HasFixAvailable: false},
+			{CVE: "m", Severity: "MEDIUM", HasFixAvailable: true},
+			{CVE: "m_nofix", Severity: "MEDIUM", HasFixAvailable: false},
+			{CVE: "h", Severity: "HIGH", HasFixAvailable: true},
+			{CVE: "h_nofix", Severity: "HIGH", HasFixAvailable: false},
+			{CVE: "c", Severity: "CRITICAL", HasFixAvailable: true},
+			{CVE: "c_nofix", Severity: "CRITICAL", HasFixAvailable: false},
 		},
 	}
-	mc := &testutil.MockMetadataClient{
-		Vulnz: []metadata.Vulnerability{
-			{CVE: "l", Severity: "LOW"},
-			{CVE: "m", Severity: "MEDIUM"},
-			{CVE: "c", Severity: "CRITICAL"},
-		},
+	var tests = []struct {
+		name                      string
+		maxSeverity               string
+		maxFixUnavailableSeverity string
+		want                      []string
+	}{
+		{"default to allow all", "", "", []string{}},
+		{"critical", "CRITICAL", "", []string{}}, // same as allow all.
+		{"high", "HIGH", "", []string{"c"}},
+		{"medium", "MEDIUM", "", []string{"h", "c"}},
+		{"low", "LOW", "", []string{"m", "h", "c"}},
+		{"BLOCK_ALL", "BLOCK_ALL", "", []string{"l", "m", "h", "c"}},
+		{"BLOCK_ALL + explicit allow unfixable", "BLOCK_ALL", "ALLOW_ALL", []string{"l", "m", "h", "c"}},
```

Comment:
um shdnt all vuln be returned?

Reply:
Not in this case, because it's explicitly allowing all unfixable vulnerabilities through.

Reply:
Fixed by giving tests better names. I see now that they were confusing.